tests/common.go

```diff
@@ -7213,10 +7213,36 @@ func EndPxBackupTorpedoTest(contexts []*scheduler.Context) {
 	if TestRailSetupSuccessful && CurrentTestRailTestCaseId != 0 && RunIdForSuite != 0 {
 		AfterEachTest(contexts, CurrentTestRailTestCaseId, RunIdForSuite)
 	}
+
 	ginkgoTestDescr := ginkgo.CurrentGinkgoTestDescription()
 	if ginkgoTestDescr.Failed {
 		log.Infof(">>>> FAILED TEST: %s", ginkgoTestDescr.FullTestText)
 	}
+
+	// Cleanup all the namespaces created by the testcase
+	err := DeleteAllNamespacesCreatedByTestCase(CurrentTestRailTestCaseId)
+	if err != nil {
+		log.Errorf("Error in deleting namespaces created by the testcase. Err: %v", err.Error())
+	}
+
+	err = SetDestinationKubeConfig()
+	if err != nil {
+		log.Errorf("Error in setting destination kubeconfig. Err: %v", err.Error())
+		return
+	}
+
+	err = DeleteAllNamespacesCreatedByTestCase(CurrentTestRailTestCaseId)
+	if err != nil {
+		log.Errorf("Error in deleting namespaces created by the testcase. Err: %v", err.Error())
+	}
+
+	err = SetSourceKubeConfig()
+	if err != nil {
+		log.Errorf("Error in setting source kubeconfig. Err: %v", err.Error())
+		return
```

Comment:
Can we remove this since we already have used defer ?

Reply:
ok

Reply:
I think we can remove this and keep only defer because defer will run no matter what

Reply:
There is no way for us to figure out why it failed if we just do defer my understanding of defer is it should be used as a fail safe if this code fails to execute. Let me know if my understanding is wrong.

Reply:
If you see the function `EndPxBackupTorpedoTest`, we are not returning anywhere. We are just catching the error and printing it.
If you remove the above line and just use defer like this, it will achieve what we want

```
defer func() {
			err := SetSourceKubeConfig()
			log.FailOnError(err, "failed to switch context to source cluster")
		}()
```

Reply:
Okay this can be done.

Reply:
Done